packages/holocron/src/reactStreaming.js

```diff
@@ -0,0 +1,39 @@
+import { createContext, useContext } from 'react';
+
+const ReactStreamingContext = createContext({});
+
+const ModuleContext = createContext();
+
+const useAsyncModuleData = (key) => {
+  const { moduleName } = useContext(ModuleContext);
+  const streamContext = useContext(ReactStreamingContext);
+  const streamedPromise = streamContext[moduleName]?.[key];
+  if (
+    streamedPromise
+      && streamedPromise instanceof Promise
+      && !streamedPromise.data
+      && !streamedPromise.error
+  ) {
+    streamedPromise
+      .then((data) => {
+        streamedPromise.data = data;
+      })
+      .catch((error) => {
+        streamedPromise.error = error;
+      });
+
+    throw streamedPromise;
+  }
```

Comment:
shouldn't this be wrapped in a `useEffect`?

Reply:
no! this has to run  on the server, and `useEffect` wont do that